File: src/main/java/com/core/base/corebase/domain/review/SurveyPeriod.java
package com.core.base.corebase.domain.review;

import jakarta.persistence.Embeddable;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Getter;
import lombok.NoArgsConstructor;

import java.time.LocalDate;

@Getter
@Builder
@Embeddable
@NoArgsConstructor
@AllArgsConstructor
public class SurveyPeriod {
    private LocalDate surveyStartDate;
    private LocalDate surveyEndDate;

    public boolean isBefore(LocalDate date) {
        return surveyStartDate.isBefore(date);
    }

    public boolean between(LocalDate date) {
        return surveyStartDate.isEqual(date) ||
               surveyEndDate.isEqual(date) ||
               (surveyStartDate.isBefore(date) && surveyEndDate.isAfter(date));
    }
}

File: src/main/kotlin/com/core/base/corebase/domain/review/SurveyPeriod.kt
package com.core.base.corebase.domain.review

import jakarta.persistence.Embeddable
import lombok.Builder
import java.time.LocalDate


@Builder
@Embeddable
class SurveyPeriod(
    val surveyStartDate: LocalDate,
    val surveyEndDate: LocalDate,
){
    fun isBefore(now: LocalDate) : Boolean = surveyStartDate.isBefore(now)

    fun between(now : LocalDate) : Boolean =
        surveyStartDate.isEqual(now)
                  || surveyEndDate.isEqual(now)
                  || (surveyStartDate.isAfter(now) && surveyEndDate.isBefore(now))

}